src/QuadForm/Lattices.jl

```diff
@@ -1471,20 +1489,21 @@ end
 #
 ################################################################################
 
-# bugged needs intersections of non-full modules to work first
 @doc Markdown.doc"""
-    _orthogonal_complement(M::AbsLat, L::AbsLat) -> AbsLat
+    orthogonal_submodule(L::AbsLat, M::AbsLat) -> AbsLat
 
-Return the orthogonal complement of the lattice `L` inside the lattice `M`.
+Return the largest submodule of `L` orthogonal to `M`.
 """
-function _orthogonal_complement(M::AbsLat, L::AbsLat)
-  @req ambient_space(M) == ambient_space(L) "lattices must be in the same ambient space"
+orthogonal_submodule(L::AbsLat, M::AbsLat) = orthogonal_submodule(L, M)
+
+function orthogonal_submodule(L::T, M::T) where T <: Union{HermLat, QuadLat}
+  @req ambient_space(M) == ambient_space(L) "Lattices must be in the same ambient space"
   V = ambient_space(L)
-  M = basis_matrix_of_rational_span(M)
-  Morth = orthogonal_complement(V, M)
-  # Now intersect KM with L
-  pm = _intersection_modules(pseudo_matrix(Morth), pseudo_matrix(L))
-  return lattice(V, pm)
+  EM = basis_matrix_of_rational_span(M)
+  Morth = orthogonal_complement(V, EM)
+  N = lattice(V, Morth)
+  N = saturate(N)
+  return intersect(L, N)
```

Comment:
I think this won't work in some cases. For instance if `L` does not have integer coordinates. 
Also fails if `M` is degenerate since then it does not have an orthogonal complement (or at least `orthogonal_complement(V,EM)` should fail).

Reply:
Don't we assume all our lattices to be non-degenerate, currently ? So what should be done instead, should I add for the moment an input check to ensure that the lattice `L` is integral ?

Reply:
No, we do not assume this (though some things will break). That is the point of calling the function `orthogonal_submodule` and not `orthogonal_complement`.  
If you implement `primitive_closure`, you can use this to compute the `orthogonal_submodule` from the corresponding orthogonal subspace.

Reply:
I made changes regarding what you said: is it better now or still incomplete/not quite true ?

Reply:
Looks pretty good. Thank you :-)